client/network/test/src/lib.rs

```diff
@@ -1064,8 +1064,10 @@ where
 		self.mut_peers(|peers| {
 			for (i, peer) in peers.iter_mut().enumerate() {
 				trace!(target: "sync", "-- Polling {}: {}", i, peer.id());
-				if let Poll::Ready(()) = peer.network.poll_unpin(cx) {
-					panic!("NetworkWorker has terminated unexpectedly.")
+				{
+					let net_poll_future = peer.network.run();
+					pin_mut!(net_poll_future);
+					let _ = net_poll_future.poll(cx);
```

Comment:
```suggestion
					peer.network.next_action().poll(cx);

```

Reply:
Doesn't compile, because the future must be `Unpin` to be polled.